lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -39,11 +39,17 @@ LinearEncodingAttr toLinearEncoding(Attribute layout, ArrayRef<int64_t> shape) {
 }
 
 unsigned getTotalElemsPerThread(Attribute layout, ArrayRef<int64_t> shape) {
+  if (auto distLayout = mlir::dyn_cast<DistributedEncodingTrait>(layout)) {
+    return distLayout.getTotalElemsPerThread(shape);
+  }
   return toLinearEncoding(layout, shape).getTotalElemsPerThread(shape);
```

Comment:
it feels like this path is not reachable, can we remove it

Reply:
Done, if you mean the one where layout is not a `DistributedEncodingTrait`.